docs/dataset/object-detection.md

```diff
@@ -0,0 +1,366 @@
+    ```
+
+    The expected CSV format for model results should look like the CSV below:
+    ```
+    locator,count_TP,count_FP,count_FN,TP,FP,FN,custom_information
+    s3://my-images/intersection_1.png,2,0,0,"[{""top_left"": [538.5, 9.0], ... }, {""top_left"": ... }]","[]","[]",0.88
+    s3://my-images/intersection_2.png,0,0,1,"[]","[]","[{""top_left"": ... }]",0.5
+    ...
+    ```
+
+    By using `locator` as the unique identifier for images, Kolena can join the model results in the CSV to the correct
+    datapoint in the dataset.
+
+    See the [`object_detection_2d/upload_results.py`](https://github.com/kolenaIO/kolena/blob/trunk/examples/dataset/object_detection_2d/object_detection_2d/upload_results.py)
+    script in the code example for details on how results were generated.
+
+=== "Web App"
+
+    To upload new model results, from the `Details` tab of the dataset, click on `Upload Model Results`
+    in the upper right.
+    Then, select `Upload From Cloud Storage`. Using the explorer, navigate to
+    `s3://kolena-public-examples/coco-2014-val/person-detection/results/` and select the `yolo_x` CSV.
+
+    You will now see a preview of how Kolena will ingest the model results. Give your model a name,
+    and click `Import` to upload the model results.
+
+    <figure markdown>
+    ![Model Results Upload](../assets/images/task-od-upload-results-light.gif#only-light)
+    ![Model Results Upload](../assets/images/task-od-upload-results-dark.gif#only-dark)
+    <figcaption markdown>Model Results Upload</figcaption>
+    </figure>
+
+    It's also important to note down the evaluation configuration of the model being uploaded, for example:
+    `iou_threshold: 0.5` or `iou_threshold: 0.3`.
+
+    <div class="grid" markdown>
+    <div markdown>
+    <figure markdown>
+        ![Example of a Model Configuration](../assets/images/task-od-model-config-dark.png#only-dark)
+        ![Example of a Model Configuration](../assets/images/task-od-model-config-light.png#only-light)
+        <figcaption>Example of a Model Configuration</figcaption>
+    </figure>
+    </div>
+    <figure markdown>
+    ![Setting Evaluation Configurations](../assets/images/task-od-evaluation-config-dark.gif#only-dark)
+    ![Setting Evaluation Configurations](../assets/images/task-od-evaluation-config-light.gif#only-light)
+    <figcaption>Setting Evaluation Configurations</figcaption>
+    </figure>
+    </div>
+
+    You can repeat the above steps with all the other model files availible.
+
+=== "SDK"
+
+    The example code contains a script [`object_detection_2d/upload_results.py`](https://github.com/kolenaIO/kolena/blob/trunk/examples/dataset/object_detection_2d/object_detection_2d/upload_results.py)
+    which will process raw model results from `s3://kolena-public-examples/coco-2014-val/transportation/results/raw/`
+    and upload model results using the `upload_object_detection_results` function.
```

Comment:
should link this to [the API doc](https://docs.kolena.com/reference/pre-built/object-detection-2d/?h=upload_object_detection_results#kolena._experimental.object_detection.upload_object_detection_results)